diff --git i/src/bind_libdatachannel.cpp w/src/bind_libdatachannel.cpp
index 1bf42b6..69de422 100644
--- i/src/bind_libdatachannel.cpp
+++ w/src/bind_libdatachannel.cpp
@@ -1217,15 +1217,15 @@ void bind_datachannel(nb::module_& m) {
       .def("is_open", &DataChannel::isOpen)
       .def("is_closed", &DataChannel::isClosed)
       .def("max_message_size", &DataChannel::maxMessageSize)
-      .def("close", &DataChannel::close)
+      .def("close", &DataChannel::close, nb::call_guard<nb::gil_scoped_release>())
       .def("send", nb::overload_cast<message_variant>(&DataChannel::send),
-           "data"_a)
+           "data"_a, nb::call_guard<nb::gil_scoped_release>())
       .def(
           "send",
           [](DataChannel& self, std::vector<byte> data, size_t size) {
             return self.send(data.data(), size);
           },
-          "data"_a, "size"_a)
+          "data"_a, "size"_a, nb::call_guard<nb::gil_scoped_release>())
       .def("stream", &DataChannel::stream)
       .def("id", &DataChannel::id)
       .def("label", &DataChannel::label)
@@ -1240,30 +1240,30 @@ void bind_track(nb::module_& m) {
       .def("is_open", &Track::isOpen)
       .def("is_closed", &Track::isClosed)
       .def("max_message_size", &Track::maxMessageSize)
-      .def("close", &Track::close)
-      .def("send", nb::overload_cast<message_variant>(&Track::send), "data"_a)
+      .def("close", &Track::close, nb::call_guard<nb::gil_scoped_release>())
+      .def("send", nb::overload_cast<message_variant>(&Track::send), "data"_a, nb::call_guard<nb::gil_scoped_release>())
       .def(
           "send",
           [](Track& self, std::vector<byte> data, size_t size) {
             return self.send(data.data(), size);
           },
-          "data"_a, "size"_a)
+          "data"_a, "size"_a, nb::call_guard<nb::gil_scoped_release>())
       .def("send_frame",
            nb::overload_cast<binary, FrameInfo>(&Track::sendFrame), "data"_a,
-           "info"_a)
+           "info"_a, nb::call_guard<nb::gil_scoped_release>())
       .def(
           "send_frame",
           [](Track& self, std::vector<byte> data, size_t size, FrameInfo info) {
             return self.sendFrame(data.data(), size, info);
           },
-          "data"_a, "size"_a, "info"_a)
+          "data"_a, "size"_a, "info"_a, nb::call_guard<nb::gil_scoped_release>())
       .def("mid", &Track::mid)
       .def("direction", &Track::direction)
       .def("description", &Track::description)
       .def("set_description", &Track::setDescription, "description"_a)
       .def("on_frame", &Track::onFrame, "callback"_a)
-      .def("request_keyframe", &Track::requestKeyframe)
-      .def("request_bitrate", &Track::requestBitrate, "bitrate"_a)
+      .def("request_keyframe", &Track::requestKeyframe, nb::call_guard<nb::gil_scoped_release>())
+      .def("request_bitrate", &Track::requestBitrate, "bitrate"_a, nb::call_guard<nb::gil_scoped_release>())
       .def("set_media_handler", &Track::setMediaHandler, "handler"_a.none())
       .def("chain_media_handler", &Track::chainMediaHandler, "handler"_a)
       .def("get_media_handler", &Track::getMediaHandler);
