diff --git i/CMakeLists.txt w/CMakeLists.txt
index ba0479b..9ddd8b5 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -19,6 +19,8 @@ message(STATUS "Dependencies directory: ${DEPS_DIR}")
 # Python / nanobind
 find_package(Python REQUIRED COMPONENTS Interpreter Development.Module)
 find_package(nanobind CONFIG REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(NICE REQUIRED IMPORTED_TARGET nice)
 
 # 依存関係用の ExternalProject
 include(ExternalProject)
@@ -134,7 +136,7 @@ else()
       -DUSE_MBEDTLS=ON
       -DMbedTLS_ROOT=${MBEDTLS_INSTALL_DIR}
       -DUSE_GNUTLS=OFF
-      -DUSE_NICE=OFF
+      -DUSE_NICE=ON
       -DNO_TESTS=ON
       -DNO_EXAMPLES=ON
       -DCMAKE_POSITION_INDEPENDENT_CODE=ON
@@ -143,7 +145,6 @@ else()
     INSTALL_COMMAND ${CMAKE_COMMAND} --build ${LIBDATACHANNEL_BUILD_DIR} --config Release --target install
     BUILD_BYPRODUCTS
       ${LIBDATACHANNEL_LIB}
-      ${LIBDATACHANNEL_INSTALL_DIR}/lib/libjuice.a
       ${LIBDATACHANNEL_INSTALL_DIR}/lib/libsrtp2.a
       ${LIBDATACHANNEL_INSTALL_DIR}/lib/libusrsctp.a
     DEPENDS mbedtls_build
@@ -182,7 +183,7 @@ if(WIN32)
 else()
   target_link_libraries(libdatachannel_ext PRIVATE
     ${LIBDATACHANNEL_LIB}
-    ${LIBDATACHANNEL_INSTALL_DIR}/lib/libjuice.a
+    PkgConfig::NICE
     ${LIBDATACHANNEL_INSTALL_DIR}/lib/libsrtp2.a
     ${LIBDATACHANNEL_INSTALL_DIR}/lib/libusrsctp.a
     ${MBEDTLS_LIB}
